src/OrchardCore.Modules/OrchardCore.Contents/Workflows/Activities/ContentForEachTask.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.Extensions.Localization;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement.Workflows;
+
+namespace OrchardCore.Contents.Workflows.Activities
+{
+    public class ContentForEachTask : TaskActivity
+    {
+        readonly IStringLocalizer S;
+        private readonly ISession _session;
+        private readonly IContentManager _contentManager;
+
+        public ContentForEachTask(IContentManager contentManager, IStringLocalizer<RetrieveContentTask> localizer, ISession session)
+        {
+            S = localizer;
+            _session = session;
+            _contentManager = contentManager;
+        }
+        public override string Name => nameof(ContentForEachTask);
+
+        public override LocalizedString DisplayText => S["Content For Each Task"];
+
+        public override LocalizedString Category => S["Content"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Iterate"], S["Done"]);
+        }
+        public override bool CanExecute(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return true;
+        }
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var contentItems = await _session
+                .Query<ContentItem, ContentItemIndex>(index => index.ContentType == ContentType && index.Published == true && index.Latest == true)
+                .ListAsync();
+            List<ContentItem> items = contentItems.ToList();
+            if (contentItems.Count() == 0)
```

Comment:
If there is no content item of this type it is throwing an Exception?

Reply:
Correct, all other content Tasks throw exceptions when content can't be found so was following the module conventions. The below is from RetriveContentTask. I do agree the workflow error handling could be better.
![image](https://user-images.githubusercontent.com/89110166/172959769-37efc54a-dcaa-4247-b384-3d0234b7c671.png)

Happy to change based on your advice.

Reply:
But it's a for loop, so it doesn't mean you expect a specific item, might be different semantic it.

Reply:
I get what you mean now, I will implement the exception on only incorrect contenttype and have a select for contenttype.